src/ResourceManager/DataLakeStore/Commands.DataLakeStore/Commands/GetAzureRmDataLakeStoreItemContent.cs

```diff
@@ -16,86 +16,133 @@
+            {
+                ConfirmAction(
+                    Resources.DownloadFileDataMessage,
+                    Path.TransformedPath,
+                    () =>
                     {
-                        Length = (long)DataLakeStoreFileSystemClient.GetFileStatus(Path.TransformedPath, Account).Length - Offset;
-                        if (Length > 1 * 1024 * 1024 && !Force)
+                        byte[] byteArray;
+                        if (Length <= 0)
+                        {
+                            Length = (long)DataLakeStoreFileSystemClient.GetFileStatus(Path.TransformedPath, Account).Length - Offset;
+                            if (Length > 1 * 1024 * 1024 && !Force)
                         // If content is greater than 1MB throw an error to the user to let them know they must pass in a length to preview this much content
                         {
-                            throw new InvalidOperationException(string.Format(Resources.FilePreviewTooLarge, 1 * 1024 * 1024, Length));
+                                throw new InvalidOperationException(string.Format(Resources.FilePreviewTooLarge, 1 * 1024 * 1024, Length));
+                            }
                         }
-                    }
 
-                    using (var memStream = ((MemoryStream)DataLakeStoreFileSystemClient.PreviewFile(Path.TransformedPath, Account, Length, Offset,
-                        CmdletCancellationToken, this)))
-                    {
-                        byteArray = memStream.ToArray();
-                    }
+                        using (var memStream = ((MemoryStream)DataLakeStoreFileSystemClient.PreviewFile(Path.TransformedPath, Account, Length, Offset,
+                            CmdletCancellationToken, this)))
+                        {
+                            byteArray = memStream.ToArray();
+                        }
 
-                    if (UsingByteEncoding(Encoding))
-                    {
-                        WriteObject(byteArray);
-                    }
-                    else
-                    {
-                        WriteObject(BytesToString(byteArray, Encoding));
-                    }
-                });
+                        if (UsingByteEncoding(Encoding))
+                        {
+                            WriteObject(byteArray);
+                        }
+                        else
+                        {
+                            WriteObject(BytesToString(byteArray, Encoding));
+                        }
+                    });
+            }
+            else
+            {
+                if(Head > 0 && Tail > 0)
+                {
+                    throw new InvalidOperationException(string.Format(Resources.HeadAndTailTogetherError, Head, Tail));
+                }
+
+                if (Head < 1 && Tail < 1)
+                {
+                    throw new InvalidOperationException(string.Format(Resources.HeadAndTailTogetherError, Head, Tail));
```

Comment:
consider this.ThrowTerminatingError()
Helps in specifying error category, which helps script writer who have $errorView="CategoryView"